samples/c2cr/parser.cr

```diff
@@ -321,8 +321,17 @@ module C2CR
           members_count += 1
 
           case c.kind
+          when .union_decl?
+              visit_union_in_struct(c, spelling)
           when .field_decl?
-            str.puts "    #{c.spelling.underscore} : #{Type.to_crystal(c.type)}"
+              field_type = Type.to_crystal(c.type)
+              case field_type.to_s
+              when .starts_with?("ANONYMOUS_")
+                  str.puts "    #{c.spelling.underscore} : #{field_type}_#{c.spelling.underscore}"
+                  puts "  alias #{field_type}_#{c.spelling.underscore} = ANONYMOUS_#{c.type.cursor.semantic_parent.spelling}_#{c.type.cursor.hash}"
```

Comment:
I think the global `puts` to generate an alias isn't a good idea. It would be better to use either the aliased name (`pkcs7_st_d`) or the unstable name (`pkcs7_st_1059675872`) directly, if possible.

Reply:
> global puts to generate an alias isn't a good idea

the global `puts` is used all over the place in this project.

> It would be better to use either the aliased name (pkcs7_st_d) or the unstable name (pkcs7_st_1059675872) directly, if possible.

Yeah, i would agree on the first one. However i could not find a way to access the parent parent parent cursor to get the field name `d`. That is why i put it as the cursor hash value (which i could get at another place). If you have a suggestion how to do that, i can add it.

On the second point, no way man. AFAIK this tool is to generate the basic LibC => Crystal definitions, so one can write a proper wrapper with classes/functions/whatevers in a nice way. The hashvalue changes on every run of clang which means the wrapper lib author would need to redefine every run of this application.

The alias is simply there so you can run `LibC::pkcs7_st_d.new` regardless of how many times you re-run this automatic definition generator.

Sorry for being a n00b at clang AST parsing.

Reply:
> the global puts is used all over the place in this project.

Oops, I didn't touch it for a while. It's fine then!

Thanks for the explanation.